README.md

```diff
@@ -44,6 +44,9 @@ $database = InfluxDB\Client::fromDSN(sprintf('influxdb://user:pass@%s:%s/%s', $h
 $client = $database->getClient();
 ```
 
+**Important:** don't forget to `urlencode()` the password (and username for that matter) when using a DSN,
```

Comment:
Is it important now that you added the encoding function? The password should not be encoded now because we are making it inside the function itself.

Reply:
It is; the username and password values are url**de**coded in the `fromDSN` method, since they're expected to be urlencoded. Passing a DSN with unencoded parts to `parse_url` might cause it to return false. So all parts of the DSN should be urlencoded, which in practice would only be an issue for the password, since that is the part of the DSN that is most likely to contain non-alphanumeric characters. The `parse_url` method only breaks down the URL into its separate parts but does not urldecode them, hence the urldecode() functions on the username and password parts when passing them to the `Client` constructor in the fromDSN method.

Now I've come to think of it, this might actually be a BC change in very specific cases. Should a password contain `%25`, this would break after this change, since that would be decoded to `%`. Consider the following code:

```php
$urls = [
  'protocol://username:pa%ss@hostname:123/path',
  'protocol://username:pa%25ss@hostname:123/path',
  'protocol://username:pa%2525ss@hostname:123/path',
];

foreach ($urls as $url) {
  $password = parse_url($url, PHP_URL_PASS);
  printf('%s / %s'.PHP_EOL, $password, urldecode($password));
}
```

This results in the following output:
```
pa%ss / pa%ss
pa%25ss / pa%ss
pa%2525ss / pa%25ss
```

So, if the password for the influx user is `pa%25ss`, it should now be written `pa%2525ss`. PHP only seems to consider a URL malformed if there's an unencoded `#` in any of its parts. An unencoded `%` will still result in the URL to be successfully parsed and will be left alone by the `urldecode` function.

But without this change, it is impossible to have a hash in the password.

Reply:
Right, thanks for your explanation!